presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -922,7 +924,9 @@ private void commitShared()
             // Partition directory must be put back before relevant metastore operation can be undone
             committer.undoAlterPartitionOperations();
 
-            rollbackShared();
+            if (!skipStagingCleanupOnFailure) {
+                deleteStagingData();
```

Comment:
Should this only be skipped when write mode is `STAGE_AND_MOVE_TO_TARGET_DIRECTORY` ? If so, one way is to make `rollbackShared()` take a boolean `skipStagingCleanup`.

Reply:
We want to skip it specifically for DIRECT_TO_TARGET_NEW_DIRECTORY.  In fact maybe that's the only case we want to clean up?

For STAGE_AND_MOVE_TO_TARGET_DIRECTORY I think we'd actually want to not add the  "cleanupTasksforAbort", but I don't quite understand why that's a separate mechanism

 For DIRECT_TO_EXISTING_DIRECTORY, we don't want to do anything.  

I think I'll limit this change to DIRECT_TO_TARGET_NEW_DIRECTORY and rename the property accordingly

Reply:
Yes, i think it is reasonable to allow `skipStagingCleanupOnFailure` to be set to `true` only if `STAGE_AND_MOVE_TO_TARGET_DIRECTORY` is seletected. If some other mode is selected - we should simply throw saying that configuration is not supported.

Reply:
Not sure if the name should be changed though

Reply:
@arhimondr We specifically need this behavior for DIRECT_TO_TARGET_NEW_DIRECTORY.

Reply:
Sorry, misscopied. Yes, make it available only for `DIRECT_TO_TARGET_NEW_DIRECTORY`.